tools/walletextension/rpcapi/wallet_extension.go

```diff
@@ -270,7 +271,7 @@ func (w *Services) Version() string {
 }
 
 func (w *Services) GetTenNodeHealthStatus() (bool, error) {
-	res, err := withPlainRPCConnection[bool](w, func(client *gethrpc.Client) (*bool, error) {
+	res, err := withPlainRPCConnection[bool](context.Background(), w, func(client *gethrpc.Client) (*bool, error) {
```

Comment:
Maybe worth putting a timeout on this context too in case the conn pool is gummed up?